crates/bevy_pbr/src/light.rs

```diff
@@ -568,19 +803,145 @@ pub(crate) fn assign_lights_to_clusters(
 
     let light_count = lights.len();
     let mut global_lights_set = HashSet::with_capacity(light_count);
-    for (view_entity, view_transform, camera, frustum, mut clusters) in views.iter_mut() {
+    for (view_entity, view_transform, camera, frustum, config, mut clusters) in views.iter_mut() {
+        if matches!(config, ClusterConfig::None) {
+            commands.entity(view_entity).remove::<VisiblePointLights>();
+            continue;
+        }
+
         let view_transform = view_transform.compute_matrix();
         let inverse_view_transform = view_transform.inverse();
-        let cluster_count = clusters.aabbs.len();
         let is_orthographic = camera.projection_matrix.w_axis.w == 1.0;
+
+        let screen_size_u32 = camera.target.get_physical_size(&windows, &images);
+        if screen_size_u32.is_none() {
+            continue;
+        }
+        let screen_size_u32 = screen_size_u32.unwrap();
+        let mut cluster_dimensions = config.dimensions_for_screen_size(screen_size_u32);
+
+        let far_z = match config.far_z_mode() {
+            ClusterFarZMode::CameraFarPlane => camera.far,
+            ClusterFarZMode::MaxLightRange => {
+                let inverse_view_row_2 = inverse_view_transform.row(2);
+                lights
+                    .iter()
+                    .map(|light| {
+                        -inverse_view_row_2.dot(light.translation.extend(1.0)) + light.range
+                    })
+                    .reduce(f32::max)
+                    .unwrap_or(0.0)
+            }
+            ClusterFarZMode::Constant(far) => far,
+        };
+        let first_slice_depth = match cluster_dimensions.z {
+            1 => config.first_slice_depth().max(far_z),
+            _ => config.first_slice_depth(),
+        };
+        // NOTE: Ensure the far_z is at least as far as the first_depth_slice to avoid clustering problems.
+        let far_z = far_z.max(first_slice_depth);
+
         let cluster_factors = calculate_cluster_factors(
-            // NOTE: Using the special cluster near value
-            clusters.near,
-            camera.far,
-            clusters.axis_slices.z as f32,
+            first_slice_depth,
+            far_z,
+            cluster_dimensions.z as f32,
             is_orthographic,
         );
 
+        let max_indices = ViewClusterBindings::MAX_INDICES;
+
+        if config.dynamic_resizing()
+            && max_indices
+                < (cluster_dimensions.x * cluster_dimensions.y * cluster_dimensions.z) as usize
+                    * light_count
+        {
```

Comment:
I rather meant to do the automatic reconfiguration always if `dynamic_resizing()` is configured:
```suggestion
        if config.dynamic_resizing() {
```
is there any reason not to? I can add a check for storage buffer support and skip this when I rebase that PR on top of this.

Reply:
ok, i guess the estimation process is not expensive so no real need to avoid it.